service/src/main/java/uk/gov/hmcts/reform/fpl/service/RespondentService.java

```diff
@@ -1,92 +1,98 @@
-    public AboutToStartOrSubmitCallbackResponse addHiddenValues(CaseDetails caseDetails) {
-        Map<String, Object> data = caseDetails.getData();
-
-        if (caseDetails.getData().containsKey("respondents1")) {
-            List<Map<String, Object>> respondentParties = (List<Map<String, Object>>) data.get("respondents1");
-
-            List<RespondentParty> respondentPartyList = respondentParties.stream()
-                .map(entry -> mapper.convertValue(entry.get("value"), Map.class))
-                .map(map -> mapper.convertValue(map.get("party"), RespondentParty.class))
-                .map(respondent -> {
-                    RespondentParty.RespondentPartyBuilder partyBuilder = respondent.toBuilder();
-
-                    if (respondent.getPartyId() == null) {
-                        partyBuilder.partyId(UUID.randomUUID().toString());
-                        partyBuilder.partyType(PartyType.INDIVIDUAL);
-                    }
-
-                    return partyBuilder.build();
-                })
-                .collect(toList());
-
-            List<Map<String, Object>> respondents = respondentPartyList.stream()
-                .map(item -> ImmutableMap.<String, Object>builder()
-                    .put("id", UUID.randomUUID().toString())
-                    .put("value", ImmutableMap.of(
-                        "party", mapper.convertValue(item, Map.class),
-                        "leadRespondentIndicator", "No"))
-                    .build())
-                .collect(toList());
-
-            data.put("respondents1", respondents);
-        }
+    private Element<Respondent> getElementToAdd(List<Element<Respondent>> confidentialChildren,
+                                                Element<Respondent> element) {
+        return confidentialChildren.stream()
+            .filter(confidentialChild -> confidentialChild.getId().equals(element.getId()))
+            .findFirst()
+            .orElse(element);
+    }
+
+    public List<Element<Respondent>> modifyHiddenValues(List<Element<Respondent>> respondents) {
+        return respondents.stream()
+            .map(element -> {
+                Respondent.RespondentBuilder builder = Respondent.builder();
+
+                if (element.getValue().getParty().getPartyId() == null) {
+                    addHiddenValues(element, builder);
+                } else {
+                    builder.party(element.getValue().getParty().toBuilder().build());
+                }
+
+                if (hiddenContactDetails(element)) {
+                    builder.party(element.getValue().getParty().toBuilder()
+                        .address(null)
+                        .telephoneNumber(null)
+                        .email(null)
+                        .build());
+                }
+
+                return Element.<Respondent>builder()
+                    .id(element.getId())
+                    .value(builder.build())
+                    .build();
```

Comment:
I was thinking if we can be clever somehow (as it duplicates with ChildService), but don't have quick idea (the builder seems to be a bit annoying here and don't want to refactor too much, but I don't like the fact it's exact copy in two places :(

Reply:
it's a carbon copy minus the builders. I can't think of a good way to make it all generic. There's no builder for the Party class and even then, hiddenDetails / contactDetailsHidden (they're different for Respondent / Children) aren't in the party class